src/app/meshes/views/MeshListView.vue

```diff
@@ -98,13 +109,38 @@
                             },
                             label: t('common.collection.actions.view'),
                           }"
+                          data-testid="dropdown-view-details-item"
                         />
                       </template>
                     </KDropdownMenu>
                   </template>
                 </AppCollection>
               </template>
             </KCard>
+
+            <RouterView
+              v-if="route.params.mesh"
+              v-slot="child"
+            >
+              <TrayView
+                @close="route.replace({
+                  name: 'mesh-list-view',
+                  params: {
+                    mesh: route.params.mesh,
+                  },
+                  query: {
+                    page: route.params.page,
+                    size: route.params.size,
+                  },
+                })"
+              >
+                <component
+                  :is="child.Component"
+                  :name="route.params.mesh"
+                  :mesh-insight="data?.items.find((item) => item.name === route.params.mesh)"
```

Comment:
If we just called this `data`/`props.data` it would probably help a little with the rollout. We mostly use `props.data` for other places where we do this sort of thing also.

Reply:
`props.data` is worse. I want to know what stuff I’m dealing with and I can by calling it `props.meshInsight`. I want to go back to all non-general examples of this and changes `props.data` there, too because it doesn’t help anything to have that general of a name while it does help to have a more specific name referring to the entity/resource.

Reply:
So this will be one of those places where we have completely opposing opinions.

Lets leave this as is for the moment. But I think even in this instance I can see plenty of problems that tell me using `props.data` is better not worse.

Briefly:

- In the case of passing these property down into a dynamic component you don't actually know whether its a MeshTrayView or maybe its something else. It just so happens that in this case it can only be a `MeshTrayView` but in the case of something like a Dataplane/Gateway list (so a list of mixed entities) we wouldn't know, therefore `data` works.
- In the case where the data used in the list can change, say for example at one point we use `meshes` for the listing and then we decided to change it to use `mesh-insights` for the listing (or vice versa) then we have to change all the property names also.

To your point of knowing what stuff is, well you are in the `MeshSummaryView`, so thats a mesh thing. Additionally You have Typescript to tell you what things are, in fact show you want things are, so you don't need to repeat that in the name of the thing.

So to be clear, lets leave this as is, but in my opinion using a `props.data` is definitely not worse, and potentially factually (see above) its not worse either.